### ISSUE TITLE
Metrics labels containing hash (#) fail to draw graphs

### ISSUE THREAD

**Issue by jessereynolds** [auxesis/visage#125]
If you include a metric identifier containing # in graph, that graph will fail to draw. Eg:

/var/lib/collectd/rrd/ns1.example.com/dns/dns_qtype-#0.rrd

The "#0" in the above identifier is the type instance. This is a valid type instance name according to https://collectd.org/wiki/index.php/Naming_schema - "...may use all ASCII characters except the null-byte and the slash character (‘/’)".

**Comment by jessereynolds** [comment]
Eg Request URL: 

```
http://stats/data/ns1.example.com/dns/dns_qtype-#99?callback=Request.JSONP.request_map.request_5
```

Returns an empty hash:
Status:200
    {}

**Comment by auxesis** [comment]
Wow, this is exceptionally dumb behaviour from Visage.

I'm am of two minds how to fix this:
- The current URL hashing structure in Visage 2.x needs to die in a fire.
  The implementation is not robust, and there is an ever increasing list of
  corner cases that make Visage difficult to use. Visage 3.0 does away with
  the hashing structure, so all effort should go into releasing 3.0
- This is a relatively straightforward bug to fix in the 2.x series, but
  we'd be flogging a dead horse.

@jessereynolds: how important is it to get this bug fixed now? Can you wait
a little while longer for 3.0?

On Friday, February 22, 2013, Jesse Reynolds wrote:

> Eg
> Request URL:http://stats/data/ns1.example.com/dns/dns_qtype-#99https://github.com/auxesis/visage/issues/99
> ?callback=Request.JSONP.request_map.request_5
> 
> Returns an empty hash:
> Status:200
> {}
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/auxesis/visage/issues/125#issuecomment-13918243.

## 

w: http://holmwood.id.au/~lindsay/
t: @auxesis

**Comment by jessereynolds** [comment]
@auxesis understand ... It's not impacting me directly so lets focus on 3.0!

**Comment by auxesis** [comment]
Awesome. I've merged the rebuilder branch into master in anticipation of the 3.0 release. 

Still needs more work, but getting closer every day.

**Comment by auxesis** [comment]
Pretty much all fixed in 3.0.

Resolving.
